### PR TITLE
taxonomy: more eurocodes for onion-family / tubers / root vegetables

### PR DESCRIPTION
**Pull request by stephanegigandet**
More Eurocodes for vegetables #8825 

Also removed some entries that are now covered by processings, and added more processings.

### CODE REVIEW

File: taxonomies/ingredients.txt [hunk 1]
```diff
@@ -64468,119 +64405,6 @@ ciqual_food_name:fr:Laurier, feuille
...
-# zh-hans:蝦夷蔥
-# zh-hant:蝦夷蔥
-# zh-tw:細香蔥
-#plant: wikidata:en:Q51148
-wikidata:en:Q61046162
-wikipedia:en:https://en.wikipedia.org/wiki/Chives
-# 4350 in 20 @2021-10-19
-
-<en:chives
-en:dehydrated chives
```
Comment (benbenben2) [inline taxonomies/ingredients.txt:64565]: -> da, nn, sv

File: taxonomies/ingredients.txt [hunk 2]
```diff
@@ -49470,1220 +49470,1223 @@ wikipedia:en:https://en.wikipedia.org/wiki/Heart_of_palm
...
+fr:oignon jaune, oignons jaunes
+nl:Gele ui
+sv:gul lök
+wikidata:en:Q8051898
+wikipedia:en:https://en.wikipedia.org/wiki/Yellow_onion
+# ingredient/fr:oignons-jaunes has 15 products @2019-05-25
 
-<fr:ail déshydraté
-<en:garlic
-en:rehydrated garlic
```
Comment (benbenben2) [inline taxonomies/ingredients.txt:50123]: -> bg, sv

File: taxonomies/ingredients.txt [hunk 3]
```diff
@@ -51163,2182 +51228,2054 @@ ciqual_food_code:en:53200
...
-fr:oignon semoule
-
-<en:onion
-fr:oignons grillés en poudre
-
-<en:onion
-fr:oignon fumé en poudre
-
-<en:onion
-en:onion purée
```
Comment (benbenben2) [inline taxonomies/ingredients.txt:51642]: -> pl (pasta)

File: taxonomies/ingredients.txt [hunk 4]
```diff
@@ -51163,2182 +51228,2054 @@ ciqual_food_code:en:53200
...
-# sco:sybae
-wikidata:en:Q774278
-wikipedia:en:https://en.wikipedia.org/wiki/Scallion
-ciqual_food_code:en:11003
-ciqual_food_name:en:Chive or spring onion, fresh
-ciqual_food_name:fr:Ciboule ou Ciboulette, fraîche
-# 3011 @2021-10-20
-
-<en:spring onion
-en:spring Onion Powder
```
Comment (benbenben2) [inline taxonomies/ingredients.txt:51717]: -> ar, da, el, nn, sv

File: taxonomies/ingredients.txt [hunk 5]
```diff
@@ -51163,2182 +51228,2054 @@ ciqual_food_code:en:53200
...
+# de:Daikonkresse
+# es:daikor cress
+# fr:pousse de daikon
+# nl:daikon cress
+# ingredient/daikon-cress has no products @2019-05-15
 
-# fr:oignons-grillés is a compound ingredient, eg:Zwiebeln gegrillt (Zwiebeln, Palmöl, Weizenmehl, Kochsalz)
 
-<en:onion
-en:Baked onion
```
Comment (benbenben2) [inline taxonomies/ingredients.txt:52078]: -> bg, pt

File: taxonomies/ingredients.txt [hunk 6]
```diff
@@ -51163,2182 +51228,2054 @@ ciqual_food_code:en:53200
...
 
-<en:potato
-en:russet potatoes
-de:Geröstete Kartoffeln
-fr:Pommes de terres rousses
+# description:fr:La SALADE est un terme générique désignant en jardinage et horticulture diverses sortes de feuilles, c'est-à-dire de plantes potagères dont les feuilles, consommées crues, entrent dans la composition d'un mets froid dont elles ont pris le nom, la « salade », sous-entendu « salade verte ».
+# comment:en:in english salad seems to refer more to the group of dishes than the group of plants.
 
-<en:potato
-en:potato powder
```
Comment (benbenben2) [inline taxonomies/ingredients.txt:53198]: -> bg, da, el, pt, sl, sv

File: taxonomies/ingredients.txt [hunk 7]
```diff
@@ -49470,1220 +49470,1223 @@ wikipedia:en:https://en.wikipedia.org/wiki/Heart_of_palm
...
-description:es:Borago officinalis, la borraja, es una especie fanerógama perteneciente a la familia de las boragináceas. La forma de cocinarla es variada, pues se emplea de diferentes maneras toda la planta, desde los tallos, hasta las hojas.
-description:fr:La Bourrache ou Bourrache officinale (Borago officinalis) est une espèce de plantes annuelles de la famille des Boraginacées. Les fleurs de bourrache et les jeunes feuilles se consomment traditionnellement à l'état frais.
-description:it:La borragine (Borago officinalis L.) è una pianta erbacea annuale della famiglia delle Boraginaceae. Le foglie giovani sono variamente impiegate in cucina.
-description:nl:De bernagie (Borago officinalis), ook vaak komkommerkruid genoemd, is een gemakkelijk herkenbare, ruwharige, eenjarige plant met blauwe bloemen. Als voedsel wordt de plant vooral in het Middellandse Zeegebied gewaardeerd als onderdeel van salades en andere zomerse gerechten.
-description:pt:Borragem (Borago officinalis L.), é uma planta herbácea anual mediterrânea. Folhas e flores são utilizadas.
-# 27 in 4 @2021-09-05
+<en:onion
+fr:oignon semoule
```
Comment (benbenben2) [inline taxonomies/ingredients.txt:49544]: can be removed?

File: taxonomies/ingredients.txt [hunk 8]
```diff
@@ -49470,1220 +49470,1223 @@ wikipedia:en:https://en.wikipedia.org/wiki/Heart_of_palm
...
-fi:punajuurimehu
-fr:jus de betterave
-hr:sok od cikle
-it:succo di barbabietola
-nl:bietensap
-no:rødbetsaft
-pl:sok z buraka, sok buraczany
-# fr:jus-de-betterave has 46 products in 3 languages @2018-10-22
+<en:garlic
+fr:Ail semoule déshydraté
```
Comment (benbenben2) [inline taxonomies/ingredients.txt:50326]: can be removed?

File: taxonomies/ingredients.txt [hunk 9]
```diff
@@ -51163,2182 +51228,2054 @@ ciqual_food_code:en:53200
...
-fr:mélange de salades, salade mélangée
-de:Mischsalat
-it:Insalata mista
-sv:salladsmix
-# ingredient/fr:mélange-de-salades has 50 products in 4 languages @2019-01-28
-# ingredient/fr:salade-mélangée has 91 products in french @2019-02-03
-# Salade mélangée (laitue, chicorée et roquette)
-# Mélange de salade  (scarole, frisée, chicorée rouge, roquette)
+<en:beetroot
+fr:concentré de betterave
```
Comment (benbenben2) [inline taxonomies/ingredients.txt:52577]: can be removed?

File: taxonomies/ingredients.txt [hunk 10]
```diff
@@ -51163,2182 +51228,2054 @@ ciqual_food_code:en:53200
...
-# azb:ساققیز اوْتو
-# vls:Schosneere
-wikidata:en:Q385259
-wikipedia:en:https://en.wikipedia.org/wiki/Scorzonera_hispanica
-ciqual_food_code:en:20197
-ciqual_food_name:en:Salsify, black, raw
-ciqual_food_name:fr:Salsifis noir, cru
-# ingredient/fr:salsifis has 58 products in 5 languages @2019-04-05
+<en:red beetroot
+fr:betterave rouge déshydratée, betterave rouge en poudre, poudre de betterave rouge
```
Comment (benbenben2) [inline taxonomies/ingredients.txt:52606]: can be removed?

File: taxonomies/ingredients.txt [hunk 11]
```diff
@@ -49470,1220 +49470,1223 @@ wikipedia:en:https://en.wikipedia.org/wiki/Heart_of_palm
...
-# zh-hant:蒜
-# zh-tw:大蒜
-wikidata:en:Q21546392
-wikipedia:en:https://en.wikipedia.org/wiki/Garlic
-# en:garlic has 22470 products in 18 languages @2018-10-23
-ciqual_food_code:en:11000
-ciqual_food_name:en:Garlic, fresh
-ciqual_food_name:fr:Ail, cru
+<en:spring onion
+pl:liofilizowana cebulka zielona
```
Comment (benbenben2) [inline taxonomies/ingredients.txt:49617]: can be removed?

**Review by benbenben2 — CHANGES_REQUESTED** [review]
Lots of moved (copy pasted content), not easy to review.

In general, I have seen ingredients containing processing methods being commented instead of deleted. I guess that it is because in some languages, for some ingredients+processing, it is causing problem (like "paste" or something with juice/concentrated-from-concentrated).

When you delete:
en:dehydrated chives
da:tørret purløg
de:Schnittlauch getrocknet
fi:kuivattu ruohosipuli
fr:ciboulette déshydratée
it:erba cipollina essiccata
nl:gedroogde bieslook
nn:tørket gressløk
pl:szczypiorek suszony
sv:torkad gräslök

It will introduce unknown ingredients for da, nn, sv as long as the logic to parse ingredient processing from the ingredient is not in Ingredients.pm. Around line ~1810 (search for "# Try to remove ingredients processing "cooked rice" -> "rice"" in the file)

On the other hand when all existing tags are in ingredients processing, the block could be deleted from ingredients.

If possible, that would be helpful for reviewing if futures PR could be split like 1 PR for eurocode, another PR for processing, another PR for reorganizing blocks (copy/paste in same file). In future PR. And if possible of course.

**Comment by stephanegigandet** [comment]
> If possible, that would be helpful for reviewing if futures PR could be split like 1 PR for eurocode, another PR for processing, another PR for reorganizing blocks (copy/paste in same file). In future PR. And if possible of course.

Yes I know it's difficult to review when the order changes. I'm changing the order in order to match the Eurocode order, so that we have a more consistent order than what we have today (entries are ordered in big chunks).

I'll refrain from removing processings in future PRs unless we cover all the languages.

File: taxonomies/ingredients.txt (hunk 8 above)
Comment (stephanegigandet) [reply taxonomies/ingredients.txt:50326]: yes, I'll remove it.

File: taxonomies/ingredients.txt (hunk 10 above)
Comment (stephanegigandet) [reply taxonomies/ingredients.txt:52606]: removing

File: taxonomies/ingredients.txt (hunk 9 above)
Comment (stephanegigandet) [reply taxonomies/ingredients.txt:52577]: removing

File: taxonomies/ingredients.txt (hunk 7 above)
Comment (stephanegigandet) [reply taxonomies/ingredients.txt:49544]: removing

File: taxonomies/ingredients.txt (hunk 3 above)
Comment (stephanegigandet) [reply taxonomies/ingredients.txt:51642]: pasta is a translation of en:paste, but the whole section is commented out for some reason

File: taxonomies/ingredients.txt (hunk 5 above)
Comment (stephanegigandet) [reply taxonomies/ingredients.txt:52078]: grelhadas is already a translation of en:grilled for Portuguese, enabled ingredients processing parsing for bg

**Comment by stephanegigandet** [comment]
@benbenben2 I enabled parsing support for ar,bg,da,fi,nb,sv, which should cover most of the processed entries that I removed.

The processing support is not perfect, we are missing translations etc. but I suggest we merge this PR first, and we can improve the parsing later, in future PRs, unrelated to the Eurocodes.

If you agree, can you approve the changes so that we can merge?

**Review by benbenben2 — APPROVED** [review]
Looks good to me.
Great work.

**Pull request merged**